tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/ElasticsearchGetTagsTests.java

```diff
@@ -0,0 +1,216 @@
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.alfresco.utility.network.ServerHealth;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@ContextConfiguration(locations = "classpath:alfresco-elasticsearch-context.xml",
+    initializers = AlfrescoStackInitializer.class)
+public class ElasticsearchGetTagsTests extends AbstractTestNGSpringContextTests
+{
+
+    @Autowired
+    private ServerHealth serverHealth;
+    @Autowired
+    private DataUser dataUser;
+    @Autowired
+    private DataSite dataSite;
+    @Autowired
+    private DataContent dataContent;
+    @Autowired
+    private RestWrapper restClient;
+
+    private UserModel user;
+    private SiteModel site;
+    private RestTagModel apple, banana, pineapple, winegrape, grapefruit, orange;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation() throws InterruptedException
+    {
+        serverHealth.assertServerIsOnline();
+
+        STEP("Create user and site");
+        user = dataUser.createRandomTestUser();
+        site = dataSite.usingUser(user).createPublicRandomSite();
+        final FileModel document = dataContent.usingAdmin().usingSite(site).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+
+        STEP("Create few tags");
+        apple = restClient.authenticateUser(user).withCoreAPI().usingResource(document)
+            .addTag(RandomData.getRandomName("apple"));
+        banana = restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI()
+            .createSingleTag(RestTagModel.builder().tag(RandomData.getRandomName("banana")).create());
+        pineapple = restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI()
+            .createSingleTag(RestTagModel.builder().tag(RandomData.getRandomName("pineapple")).create());
+        winegrape = restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI()
+            .createSingleTag(RestTagModel.builder().tag(RandomData.getRandomName("winegrape")).create());
+        grapefruit = restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI()
+            .createSingleTag(RestTagModel.builder().tag(RandomData.getRandomName("grapefruit")).create());
+        orange = restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI()
+            .createSingleTag(RestTagModel.builder().tag(RandomData.getRandomName("orange")).create());
+
+        STEP("Wait for indexing to complete");
+        Utility.sleep(500, 60000, () -> restClient.authenticateUser(dataUser.getAdminUser())
```

Comment:
I recently changed the default timeout for ES from 60s to 10s - 60s is complete overkill (even on our CI). You can see the code in the `SearchQueryService`.

Reply:
I will fix the timeout, thanks!